engine/dlib/src/dlib/opaque_handle_container.h

```diff
@@ -105,6 +105,21 @@ class dmOpaqueHandleContainer
      */
     T* GetByIndex(uint32_t index);
 
+    /*# container get handle by index
+     *
+     * Constructs an opaque handle from the index itself plus the stored version for that index.
+     * The index must be within a valid range of the backing array,
+     * as denoted by the container capacity ([0...c-1] where c equals capacity).
+     *
+     * Note: this is indented for internal use only and relies on the fact that the backing
+     *       data structure for the container is a continuous array.
+     *
+     * @name IndexToHandle
+     * @param index [type:uint32_t] The index to query the container with
+     * @return [type:HOpaqueHandle] Return the constructed object handle
+     */
+    HOpaqueHandle IndexToHandle(uint32_t index);
```

Comment:
We could potentially introduce an iterator instead and remove these "by index" functions. But for now it's fine I think?